### PR TITLE
improve README

### PR DESCRIPTION
**Pull request by timogoebel**
@leewaa: Do you mind taking a look and checking my English? Thanks.

### CODE REVIEW

**Comment by Raboo** [comment]
Perhaps it's good to move the example templates from gist into this repository?

File: README.md [hunk 1]
```diff
@@ -14,6 +14,22 @@ See [Plugins install instructions](https://theforeman.org/plugins/)
 for how to install Foreman plugins.
 You need to install the package `tfm-rubygem-foreman_userdata`.
 
+## Motivation
+
+This plug-in was developed to be used with VMWare vSphere provisioning from VMWare templates.
+vSphere offers the functionality to customize the templates during cloning. This is very handy to change the network settings of a newly cloned VM.
```
Comment (leewaa) [inline README.md]: very handy *for changing* the

**Review by leewaa — CHANGES_REQUESTED** [review]

**Comment by leewaa** [comment]
other than that, looks good to me

**Comment by dhajoshi** [comment]
Is it possible to add required network port information ? Foreman to vCenter , proxy to client.

Some prerequisite of installing cloud-init before creating template or how foreman creates it for you.

Thanks,

**Comment by timogoebel** [comment]
> Perhaps it's good to move the example templates from gist into this repository?

@Raboo: Yeah, ideally they would be seeded automatically into the app. Let's create a separate issue for that.

> Is it possible to add required network port information ? Foreman to vCenter , proxy to client.

@dhajoshi: Do you mean what ports need to be opened at the firewall for this to work?
Your hosts need to be able to access Foreman on `tcp/80` and `tcp/443`.

> Some prerequisite of installing cloud-init before creating template or how foreman creates it for you.

That should already be covered. `yum install cloud-init` when you create the template. Unfortunately, Foreman can't create it for you. But I agree that we should document some best practices to create a template.

File: README.md [hunk 2]
```diff
@@ -14,6 +14,22 @@ See [Plugins install instructions](https://theforeman.org/plugins/)
...
+The amount of things vSphere allows a user to customize are very limited, though. It's not possible to run a finish script or setup puppet with valid certificates on the new VM.
+This can be worked around by setting up a two step process: vSphere customization is just used to set up the network config of the host. Cloud-init is used to do the rest of the customization on first boot.
+
+This leads to this workflow:
+1. An administrator creates a vSphere VM template and sets up cloud-init as described below.
+2. Foreman creates a new cloned VM in vSphere and passes the [userdata template](https://gist.github.com/timogoebel/959deecff4f8ca8bcdd5fc9f95cbc8e3#file-userdata-template-erb) to vSphere.
+3. vSphere changes the network settings of the new host as defined in the userdata template and boots the new VM.
+4. The VM runs cloud-init and asks Foreman for the [cloud-init template](https://gist.github.com/timogoebel/959deecff4f8ca8bcdd5fc9f95cbc8e3#file-cloud-init-template-erb). Your VM needs network access to Foreman on ports `tcp/80` and `tcp/443`. Cloud-init then runs the template and executes the defined actions.
```
Comment (dhajoshi) [inline README.md:30]: @timogoebel , Thanks for updating this one, may be this is for RFE, but can client connect to proxy (capsule), on https (443) to get the template ? or it has to be master only and both 80 and 443 ?

Hope templates are not available at 8000 ?

Comment (timogoebel) [reply README.md:30]: This would be a RFE, `foreman_userdata` does not have a smart proxy (capsule) plugin. So this is master only.

**Pull request merged**